File: Python3/1077.py
n = int(input())
ans = ''
for i in range(n):
    a = input()
    a = a[::-1]
    length = len(a)
    if i == 0:
        ans = a
        continue
    else:
        length2 = len(ans)
        min_len = min(length,length2)
        for j in range(min_len):
            if ans[j] != a[j]:
                ans = ans[:j]
                break
ans = ans[::-1]
if(len(ans) == 0):
    print("nai")
else:
    print("%s"%ans)

File: c++/1077.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/9/22.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <cstdio>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <string.h>
using namespace std;

int main(){
    int n;
    ios::sync_with_stdio(false);
    cin.tie(0);
    scanf("%d",&n);
    scanf("%*c");
    string ans;
    for (int i=0; i<n; i++) {
        string a;
        getline(cin,a);
        reverse(a.begin(), a.end());
        int lens = a.length();
        if(i==0){
            ans = a;
            continue;
        }
        else{
            int lenans = ans.length();
            int minlen = min(lens,lenans);
            for(int j=0;j<minlen;j++){
                if(ans[j] != a[j]){
                    ans = ans.substr(0,j);
                    break;
                }
                
            }
        }
    }
    reverse(ans.begin(),ans.end());
    if(ans.size() == 0)
        printf("nai\n");
    else{
        cout << ans;
    }
}
